cmd/search_log.go

```diff
@@ -354,10 +364,91 @@ func latencyAggregateFn(cmd *cobra.Command, args []string) {
 			printCsv([][]string{sortedRecords[i]})
 			worstQueriesTrs = fmt.Sprintf("%s%s", worstQueriesTrs, printHtmlTr(sortedRecords[i], false, ""))
 		}
-		worstQueriesHtmlPart = fmt.Sprintf("<h3>%d worst queries</h3><table>%s</table>", worstQueriesPrintCnt, worstQueriesTrs)		
+		worstQueriesHtmlPart = fmt.Sprintf("<h3>%d worst queries</h3><table>%s</table>", worstQueriesPrintCnt, worstQueriesTrs)
 	}
 	finalHtml := fmt.Sprintf("%s%s", operationsHtmlPart, worstQueriesHtmlPart)
 	err = ioutil.WriteFile(latencyOutputHtml, []byte(finalHtml), 0644)
 	utils.Must(err)
 	log.Info("HTML printed.")
 }
+
+func queriesAggregateFn(cmd *cobra.Command, args []string) {
+	logger := initLogger()
+	printCsv([][]string{{"Term", "Count", "SortType", "Language"}})
+	SLICES := 100
+	RESULTS_FOR_LANGUAGE := 1000
+	gteStr := null.NewString("", false)
+	languages := []string{consts.LANG_HEBREW, consts.LANG_ENGLISH, consts.LANG_RUSSIAN} // Currently only the 3 main languages are supported due to language recognition issues
+	queryCountByLang := map[string]map[string]int{}                                     // Language -> [Search Term -> Count]
+	if lastDays > 0 {
+		gteStr = null.StringFrom(fmt.Sprintf("now-%dd/d", lastDays))
+	}
+	for i := 0; i < SLICES; i++ {
+		s := elastic.NewSliceQuery().Id(i).Max(SLICES)
+		queries, err := logger.GetLattestQueries(s, gteStr, null.NewBool(false, false))
+		utils.Must(err)
+		for _, sl := range queries {
+			term := simpleQuery(sl.Query)
+			if term == "" || sl.Error != nil || sl.QueryResult == nil || sl.Query.Deb || sl.From > 0 {
+				continue
+			}
+			lang := sl.QueryResult.(map[string]interface{})["language"].(string)
+			if lang == "" {
+				sort.Strings(sl.Query.LanguageOrder)
+				intersected := utils.IntersectSortedStringSlices(languages, sl.Query.LanguageOrder)
+				if len(intersected) > 0 {
+					lang = intersected[0]
+				}
+			}
+			if utils.Contains(utils.Is(languages), lang) {
+				if _, ok := queryCountByLang[lang]; !ok {
+					queryCountByLang[lang] = map[string]int{}
+				}
+				if _, ok := queryCountByLang[lang][term]; !ok {
+					queryCountByLang[lang][term] = 0
+				}
+				queryCountByLang[lang][term]++
+			}
+		}
+	}
+	records := [][]string{}
+	for _, lang := range languages {
+		tcs := []TermAndCount{}
+		for k, v := range queryCountByLang[lang] {
+			tcs = append(tcs, TermAndCount{Term: k, Count: v})
+		}
+		if len(tcs) < RESULTS_FOR_LANGUAGE {
+			log.Errorf("Amount of found terms (%d) for language '%s' is smaller than RESULTS_FOR_LANGUAGE const (%d).", len(tcs), lang, RESULTS_FOR_LANGUAGE)
+			return
+		}
+		sort.Slice(tcs, func(i, j int) bool {
+			return tcs[i].Count > tcs[j].Count
```

Comment:
Is this the count of queries with the same text?

Reply:
Yes. Same search term.